src/Common/Locale.php

```diff
@@ -18,13 +18,23 @@ protected function __construct(string $locale)
         $this->setLocale($locale);
     }
 
+    /**
+     * @param string $locale
+     *
+     * @return static
+     */
     public static function fromString(string $locale): self
     {
         return new static($locale);
     }
 
     abstract protected function getPossibleLanguages(): array;
 
+    /**
+     * @param string $locale
+     *
+     * @return static
```

Comment:
I don't think this PHPDoc has any useful purpose.

Reply:
it does, the static helps in the classes that extend it to show the correct type there